object_tracker/object_tracker.go

```diff
@@ -292,6 +306,42 @@ func (t *myTracker) GetObjectPointClouds(
 	return nil, errUnimplemented
 }
 
+func (t *myTracker) CaptureAllFromCamera(
+	ctx context.Context,
+	cameraName string,
+	opt viscapture.CaptureOptions,
+	extra map[string]interface{},
+) (viscapture.VisCapture, error) {
+	var detections []objdet.Detection
+	var img image.Image
+	select {
+	case <-t.cancelContext.Done():
+		return viscapture.VisCapture{}, t.cancelContext.Err()
+	case <-ctx.Done():
+		return viscapture.VisCapture{}, ctx.Err()
+	default:
+		if opt.ReturnImage {
+			if cameraName != t.camName {
+				return viscapture.VisCapture{}, errors.Errorf("Camera name given to method, %v is not the same as configured camera %v", cameraName, t.camName)
+			}
+			if opt.ReturnClassifications {
+				t.logger.Debugf("classifications requested but object tracker  %q does not implement a Detector", t.Named.Name())
+			}
+			if opt.ReturnObject {
+				t.logger.Debugf("point cloud objects requested but object tracker  %q does not return point cloud objects", t.Named.Name())
+			}
+			if opt.ReturnObject {
+				detections = t.oldDetections.Load()[1]
+			}
```

Comment:
as you pointed out, you'll want to change to `ReturnDetections` - but you will also want to take this out of the overall `if`, because a person might want the detections, but not the image

Reply:
done